### ISSUE TITLE
what is your test-driven research analysis development workflow? if you haven't adopted automated testing, why?

### ISSUE THREAD

**Issue by softloud** [softloud/neet#2]

**Comment by DrMattG** [comment]
I haven't adopted automatic testing yet. I think it is really important but I feel that I lack the technical skills to implement it in a way that I understand what it is actually doing.

**Comment by softloud** [comment]
Thanks for being brave enough to admit it!

I am hearing this a lot, but mostly sotto voce as people feel like they're
letting science down.

There's a lot of hand waving about best practice in scientific computing,
but if best practice is so dauntingly out of discipline for most
computational scientists (and give me a scientist who isn't computational
these days!), then how do we facilitate good enough scientific practice so
that's actually implemented? What would empower you to feel you had the
technical skills?

People think because I am trained in mathematics I know programming, but
all I learned to do was solve equations and write proofs. Almost all of my
work for my ugrad was with pen and paper.

I have found grasping the idea of testing not so bad, but integrating
testing into my workflow has been slow and painful. I am thinking a lot
about how to save the other members of the reproducibility team some of the
struggle.

Do let me know if you try out the workflow, it's very much a work in
progress.
